test/test_view_ops.py

```diff
@@ -457,6 +463,61 @@ def test_reshape_nonview(self, device):
         nv[6] = 0
         self.assertNotEqual(t[1, 1], nv[6])
 
+    def test_flatten_view(self, device):
+        def assert_is_view(t, v, t_is_view=False):
```

Comment:
`assert_is_view` is a misleading name for this function since it can either assert that v is a view of t or that v and t view the same base.

I suggest cutting the `is_view_of`/`is_view_of_same_base` part out and just calling those functions directly. Then this function would be something like:

```
def test_writes_propagate(t, v):
  idx_t = (0,) * t.ndim
  idx_v = (0,) * v.ndim
  v[idx_v] = 0
  self.assertEqual(t[idx_t], v[idx_v])
```

Reply:
This approach will add a second line wherever you're currently calling this function but I think that's preferable for clarity, and a function like `test_writes_propagate` may be refactored later